### PR TITLE
Dan docs

### PR DESCRIPTION
**Pull request by chillenberger**
open source docs blog and dashboard.

### CODE REVIEW

**Review by montanalow — APPROVED** [review]

File: pgml-dashboard/content/blog/data-is-living-and-relational.md [hunk 1]
```diff
@@ -21,29 +20,27 @@ img.center {
   margin: 16px 12.5%;
   max-width: 75%;
 }
-</style>
+</style> -->
```
[the remark this answers is not in the document]
Comment (chillenberger) [reply pgml-dashboard/content/blog/data-is-living-and-relational.md]: deleted

File: pgml-dashboard/src/templates/docs.rs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+use std::hash::{Hash, Hasher};
+
+use sailfish::TemplateOnce;
+
+use crate::utils::markdown::SearchResult;
+
+/// The documentation & blog template that renders markdown.
+// #[derive(TemplateOnce)]
+// #[template(path = "docs/dynamic.html")]
+// pub struct Dynamic {
```
[the remark this answers is not in the document]
Comment (chillenberger) [reply pgml-dashboard/src/templates/docs.rs]: deleted

File: pgml-dashboard/src/fairings.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+impl RequestMonitor {
+    pub fn new() -> RequestMonitor {
+        RequestMonitor {}
+    }
+}
+
+static PATH_IDS: OnceCell<regex::Regex> = OnceCell::new();
+
+#[rocket::async_trait]
+impl Fairing for RequestMonitor {
```
Comment (levkk) [inline pgml-dashboard/src/fairings.rs:23]: Don't really need this in the open source version I don't think.

File: pgml-dashboard/src/main.rs [hunk 4]
```diff
@@ -1,14 +1,121 @@
...
+pub async fn error() -> Result<(), BadRequest> {
+    info!("This is additional information for the test");
+    error!("This is a test");
+    let error: Option<i32> = None;
+    error.unwrap();
+    Ok(())
+}
+
+#[catch(403)]
+async fn not_authorized_catcher(_status: Status, _request: &Request<'_>) -> Redirect {
```
Comment (levkk) [inline pgml-dashboard/src/main.rs:26]: Don't have a `/login` in the dashboard, could remove this.

**Comment by montanalow** [comment]
🎉

**Pull request merged**